components/abi-form/abi-input.tsx

```diff
@@ -0,0 +1,72 @@
+import { cn } from "lib/utils";
+import { matchArrayType } from "lib/utils";
+import { useCallback } from "react";
+import { ArrayInput } from "./array-input";
+import { Basic } from "./basic";
```

Comment:
one thing I forgot to mention:
if you look at other components, you'll see that local files are all imported with `.js` extension, and with local paths only (i.e.: `../lib/utils` instead of `lib/utils`)

for some reason, the way you wrote works locally on this package, but fails when using these as a dependency

(I'd love to solve that but don't know how)